src/com/mendix/widget/GoogleMaps/components/__tests__/Map.spec.ts

```diff
@@ -0,0 +1,201 @@
+import { shallow } from "enzyme";
+import GoogleMap from "google-map-react";
+import { DOM, createElement } from "react";
+
+import { Map, MapProps } from "../Map";
+import { Marker } from "../Marker";
+
+import {
+    EventMock, GeocoderLocationType, GeocoderMock, GeocoderStatus, LatLngBoundsMock,
+    LatLngMock, MapsMock, MarkerMock, MockGoogle
+} from "../../../../../../../tests/mocks/GoogleMaps";
+
+import { MxMock, MxUiMock } from "../../../../../../../tests/mocks/Mendix";
+describe("Map", () => {
+    const address = "Lumumba Ave, Kampala, Uganda";
+    const invalidAddress = "invalidAddress";
+    const APIKey = "AIzaSyACjBNesZXeRFx86N7RMCWiTQP5GT_jDec";
+    const renderMap = (props: MapProps) => shallow(createElement(Map, props));
+    const defaultCenterLocation = { lat: 51.9107963, lng: 4.4789878 };
+    const successMockLocation = { lat: 30, lng: 118 };
+    const multipleAddressMockLocation = { lat: 34.213171, lng: -118.571022 };
+
+    beforeAll(() => {
+        window.google = MockGoogle;
+        window.google.maps = {};
+        window.google.maps.Geocoder = GeocoderMock;
+        window.google.maps.Map = MapsMock;
+        window.google.maps.LatLngBounds = LatLngBoundsMock;
+        window.google.maps.LatLng = LatLngMock;
+        window.google.maps.Marker = MarkerMock;
+        window.google.maps.event = EventMock;
+        window.google.maps.GeocoderStatus = GeocoderStatus;
+        window.google.maps.GeocoderLocationType = GeocoderLocationType;
+        window.mx = MxMock.prototype;
+        window.mx.ui = MxUiMock.prototype;
+    });
+
+    it("should render with the map structure", () => {
+        const map = renderMap({ address });
+        expect(map).toBeElement(
+            DOM.div({ className: "widget-google-maps" },
+                createElement(GoogleMap, {
+                    bootstrapURLKeys: { key: undefined },
+                    center: defaultCenterLocation,
+                    defaultZoom: 14,
+                    onGoogleApiLoaded: jasmine.any(Function) as any,
+                    resetBoundsOnResize: true,
+                    yesIWantToUseGoogleMapApiInternals: true
+                })
+            )
+        );
+    });
+
+    it("renders with classes", () => {
+        const googleMap = renderMap({ address });
+
+        expect(googleMap.hasClass("widget-google-maps")).toBe(true);
+    });
+
+    describe("with no address", () => {
+        it("should not look up the location", () => {
+            spyOn(window.google.maps.Geocoder.prototype, "geocode").and.callThrough();
+
+            const output = renderMap({ address: "" });
+            output.find(GoogleMap).prop("onGoogleApiLoaded")();
```

Comment:
use `.apply` at the end to make it more explicit + define google APIs before emulating this event + imho should be extracted to the function, as you need to reuse it in multiple places.